### DISCUSSION TITLE
Filtering with hnsw

### DISCUSSION THREAD

**Discussion by josippavicic** [tensorchord/pgvecto.rs#363] [Q&A] [answered]
It was my understanding that pgvecto.rs solved the problem for filtering with HNSW. I have run the docker container and created a table 
```
DROP EXTENSION IF EXISTS vectors;
CREATE EXTENSION vectors;

create table chunk (
	id uuid primary key,
	embedding vector (384) not null,
	letter varchar
);

create index on chunk using vectors (embedding vector_cos_ops);
```

I have inserted 1k demo data with letter 'A' and 1k with letter 'B'.

When I do HNSW with:
```
select * from chunk 
where chunk.letter='B'
order by embedding <=> '[0.001961352629124461, 0.0008733998021052378, ... , 0.0009683990587464525, 0.003188614714374138]'
limit 64;
```
I get only 30 instead of 64. When I run it without the where clause it gives 64, 30 with letter 'B'  and 34 with letter 'A'. It uses the HNSW index in both searches. So my conclusion is that this does postfiltering just as pgvector is. Am I doing something wrong or did I misunderstood the filtering of pgvecto.rs?

This is from pgvector vs pgvecto.rs q&a:

> Filter Support: pgvecto.rs supports VBASE search mode, ensuring precise and complete results even with applied filters and joins. pgvector may produce incomplete results when filters are applied, potentially missing relevant results

**Comment by VoVAllen** [comment, marked as answer]
Are you using 0.2.0? It should work out of the box. The docker tag in readme is old and we haven't updated it yet. You can try `docker pull tensorchord/pgvecto-rs:pg16-v0.2.0` or `docker pull tensorchord/pgvecto-rs:pg15-v0.2.0`

**Reply by josippavicic** [reply]
that was it, thanks!

**Reply by VoVAllen** [reply]
Glad to hear it works! Feel free to ask any question you met and we're ready to help.
